pkg/common/idutil/spiffeid.go

```diff
@@ -350,19 +350,6 @@ func ServerURI(trustDomain string) *url.URL {
 	}
 }
 
-// ValidateTrustDomainWorkload validates if the given SPIFFE ID
-// is a SPIFFE ID for a workload belonging to the specified
-// trust domain (e.g. spiffe://domain.test/workload)
-func ValidateTrustDomainWorkload(id spiffeid.ID, td spiffeid.TrustDomain) error {
-	if !id.MemberOf(td) {
-		return fmt.Errorf("%q does not belong to trust domain %q", id.String(), td.String())
-	}
-	if id.Path() == "" {
-		return fmt.Errorf("invalid workload SPIFFE ID %q: path is empty", id.String())
-	}
-	if IsReservedPath(id.Path()) {
-		return fmt.Errorf(`%q is not a valid workload SPIFFE ID: invalid path: "/spire/*" namespace is reserved`, id.String())
-	}
-
-	return nil
+func ServerSpiffeID(trustDomain spiffeid.TrustDomain) spiffeid.ID {
```

Comment:
That's a better name but there's already a function named `ServerID` in that package, that takes a `trustDomain` as `string` and returns a `string` and is called in a few packages across server and agent. I can make the refactor to make that function take and return the strong types, but it will expand a little bit beyond the scope of this PR, what do you think ?

Reply:
Oh, don't worry about it then. We can clean up the names towards the end of the work introducing the new types. I imagine most of the `idutil` package to be removed at that point other than some light verification related functions and other "SPIRE" specific things (e.g. `ServerID`)